src/_pytest/config/__init__.py

```diff
@@ -512,7 +512,7 @@ def _importconftest(self, conftestpath):
             _ensure_removed_sysmodule(conftestpath.purebasename)
 
         try:
-            mod = conftestpath.pyimport()
+            mod = conftestpath.pyimport(ensuresyspath=importmode)
```

Comment:
One question about this: previously this `pyimport` always used the default, even if the importmode was `append` (non-default). Now also `append` is plumbed into here, which changes the existing behavior. Does that make any difference?

Reply:
It might change things indeed, but I'm inclined that this was actually an oversight.

What do you think @RonnyPfannschmidt?